Source/Core/InputCommon/DynamicInputTextureConfiguration.cpp

```diff
@@ -0,0 +1,375 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "InputCommon/DynamicInputTextureConfiguration.h"
+
+#include <optional>
+#include <sstream>
+#include <string>
+
+#include <fmt/format.h>
+#include <picojson.h>
+
+#include "Common/CommonPaths.h"
+#include "Common/File.h"
+#include "Common/FileUtil.h"
+#include "Common/Logging/Log.h"
+#include "Common/StringUtil.h"
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+#include "InputCommon/ControllerEmu/ControllerEmu.h"
+#include "InputCommon/ImageOperations.h"
+#include "VideoCommon/RenderBase.h"
+
+namespace
+{
+std::string GetStreamAsString(std::ifstream& stream)
+{
+  std::stringstream ss;
+  ss << stream.rdbuf();
+  return ss.str();
+}
+
+std::optional<int> GetColorCodeFromString(const std::string& str)
+{
+  if (!str.empty() && str[0] == '#')
+  {
+    const std::string color_code = str.substr(1);
+    if (color_code.size() == 6)
+    {
+      int result;
+      if (TryParse(color_code, &result, 16))
+        return result;
+    }
+    else if (color_code.size() == 8)
+    {
+      // Color code includes alpha, that's currently not supported
+      int result;
+      if (TryParse(color_code.substr(2), &result, 16))
```

Comment:
Wouldn't the alpha value be at the end of the string representation (RRGGBBAA)? If so, shouldn't this be `color_code.substr(0, 6)` instead?

Reply:
This was added based off [my tool](https://github.com/iwubcode/DolphinDynamicInputTextureCreator) which uses a microsoft color struct serialization.  Json generated via that tool had AA first from what I can tell.